### PR TITLE
Add initial memory randomization

### PR DESCRIPTION
**Pull request by DrewScoggins**
Added the memory randomization as laid out by @adamsitnik in https://github.com/dotnet/performance/issues/1602

### CODE REVIEW

File: src/benchmarks/micro/libraries/System.Memory/Span.cs [hunk 1]
```diff
@@ -71,12 +77,18 @@ public void SetupIndexOf()
...
         public int IndexOfAnyTwoValues() => new System.Span<T>(_emptyWithSingleValue).IndexOfAny(_notDefaultValue, _notDefaultValue);
 
+        [MemoryRandomization(true)]
+        [IterationCount(50)]
+        [Outliers(Perfolizer.Mathematics.OutlierDetection.OutlierMode.DontRemove)]
         [Benchmark]
         public int IndexOfAnyThreeValues() => new System.Span<T>(_emptyWithSingleValue).IndexOfAny(_notDefaultValue, _notDefaultValue, _notDefaultValue);
 
         [Benchmark]
         public int IndexOfAnyFourValues() => new System.Span<T>(_emptyWithSingleValue).IndexOfAny(new ReadOnlySpan<T>(_fourValues));
```
Comment (LoopedBard3) [inline src/benchmarks/micro/libraries/System.Memory/Span.cs:87]: Is there a specific reason not to do the randomization for both?

**Review by LoopedBard3 — APPROVED** [review]
LGTM once the CI finishes.

**Comment by kunalspathak** [comment]
How are we  planning to represent these numbers in the test history graph? What happens to the existing test history?

cc: @AndyAyersMS

File: src/benchmarks/micro/libraries/System.Memory/Span.cs (hunk 1 above)
Comment (DrewScoggins) [reply src/benchmarks/micro/libraries/System.Memory/Span.cs:87]: I just went with the initial set that Adam had laid out. We might expand these going forward however.

**Comment by DrewScoggins** [comment]
Right now the plan is still use the average, with error bars, to display these tests. As for the existing test history, I didn't really believe that these tests were of much value, given how they behave with respect to memory alignment So I was not too worried about the trend changing. We could give the test a new name, or duplicate the tests and have one run with alignment and one without if we prefer.

File: src/benchmarks/micro/libraries/System.IO.FileSystem/Perf.FileStream.cs [hunk 2]
```diff
@@ -135,6 +135,9 @@ public void SeekBackward(long fileSize, FileOptions options)
             }
         }
 
+        [MemoryRandomization(true)]
+        [IterationCount(50)]
+        [Outliers(Perfolizer.Mathematics.OutlierDetection.OutlierMode.DontRemove)]
```
Comment (adamsitnik) [inline src/benchmarks/micro/libraries/System.IO.FileSystem/Perf.FileStream.cs:140]: After my recent work on `FileStream` rewrite I doubt that any of the File IO benchmarks are affected by alignment issues.

@DrewScoggins have you tried comparing the results for MemoryRandomization enabled vs disabled for this benchmark? If we can't observe any gains, we should not enable it.

```suggestion
```

**Comment by adamsitnik** [comment]
> Right now the plan is still use the average, with error bars, to display these tests.

Do you have some sample screen shots?

I was wondering if we could display every histogram bucket on the chart with a size equal to the number of iterations belonging to it.

We could use: https://plotly.com/javascript/bubble-charts/#bubble-size-scaling-on-charts. Example:

```ini
-------------------- Histogram --------------------
[ 79.009 ns ; 183.239 ns) | @@@@@@@@@@
[183.239 ns ; 287.047 ns) | @
[287.047 ns ; 391.277 ns) | @@@@@@@@
[391.277 ns ; 501.821 ns) | @
---------------------------------------------------
```

Would give us 4 points:
* for each of them, x would be the same
* y would be equal to median of values in given histogram
* size would be equal the number of iterations belonging to given bucket

We could also try box plots, but most likely we have too many data points to visualize

File: src/benchmarks/micro/libraries/System.Collections/Contains/ContainsFalse.cs [hunk 3]
```diff
@@ -37,6 +37,9 @@ public class ContainsFalse<T> where T : IEquatable<T>
         [GlobalSetup(Targets = new[] { nameof(Array), "Span" })]
         public void SetupArray() => _array = Setup();
 
+        [MemoryRandomization(true)]
+        [IterationCount(50)]
+        [Outliers(Perfolizer.Mathematics.OutlierDetection.OutlierMode.DontRemove)]
```
[the remark this answers is not in the document]
Comment (DrewScoggins) [reply src/benchmarks/micro/libraries/System.Collections/Contains/ContainsFalse.cs:42]: OK, I was just going off of the way that it was called in your PR. I will take this out.

**Comment by danmoseley** [comment]
@DrewScoggins just curious whether you plan to proceed with this one, it sounds intriguingly useful..

**Comment by DrewScoggins** [comment]
I have been fully focused on other work, that should be finishing up in a week or two. My hope was to get back this along with other requests around test execution and visualization after.

**Comment by cincuranet** [comment]
Closing in favor of https://github.com/dotnet/runtime/pull/95291.

**Pull request closed without merging**